internal/vault/vault_test.go

```diff
@@ -1169,3 +1170,71 @@ func TestRequestTokenWithAppRoleRef(t *testing.T) {
 		})
 	}
 }
+
+type testNamespacedRequestT struct {
+	expectedNamespace string
+
+	issuer *cmapi.VaultIssuer
+}
+
+func testHTTPServer(t *testing.T, handler http.Handler) (*vault.Config, net.Listener) {
```

Comment:
Can we simplify this code by replacing this function with https://pkg.go.dev/net/http/httptest#NewServer?

Reply:
indeed we can! Great suggestion 👍

Reply:
and you can also just call `Close()` on `server` instead of `server.Listener`.

Reply:
Good catch, thank you!

Reply:
sorry for all the nitpicking, but I would also use `server.URL` instead of `server.Listener.Addr()`

Reply:
no worries at all! These are all valid and good suggestions. I appreciate your keen eye :)